obspy/core/trace.py

```diff
@@ -2741,6 +3191,19 @@ def remove_response(self, inventory=None, output="VEL", water_level=60,
         self.data = data
         return self
 
+    def fft(self, window_function='cosine', percentage=0.1):
```

Comment:
I think this should rather be attached to the `TimeSeriesTrace`. I guess the `fft()` method then has to do some introspection to decide if it wants to return `FrequencyDomainTimeSeriesTrace` or  a `FrequencyDomainTrace`.